src/Symfony/Bundle/FrameworkBundle/Secrets/SecretsBroker.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Symfony\Bundle\FrameworkBundle\Secrets;
+
+class SecretsBroker extends BaseSecretsHandler
+{
+    private $checkCachedSecrets;
+    private $decryptedSecrets;
+
+    public function __construct(string $projectRoot, string $environment, bool $checkCachedSecrets = true)
+    {
+        $this->checkCachedSecrets = $checkCachedSecrets;
+        parent::__construct($projectRoot, $environment);
+    }
+
+    public function fetchSecret(string $name)
+    {
+        if ($this->checkCachedSecrets) {
```

Comment:
If perf is really not a problem, then this isn't needed (and simpler PR will get through more easily)